### PR TITLE
src/content/bootnodes: add bootnodes page

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
## Description

Adds a page about bootnodes 

## Related Issue

https://twitter.com/mysticryuujin/status/1605022656184561664?s=20&t=k9hoqPiFbileT9EXUiPuwA

cc @MysticRyuujin

### CODE REVIEW

**Comment by jmcook1186** [comment]
Hi @MariusVanDerWijden , thanks a lot for doing this for us - pinging @samajammin @wackerow @minimalsm for opinions on location - not sure `src/content/bootnodes` is best - maybe better in `developer/docs/nodes-and-clients/bootnodes` or similar?

File: src/content/bootnodes/index.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Connect to a bootnode {#connect-to-bootnode}
+
+Most clients have a list of bootnodes build in, but you might want to connect to other bootnodes to connect by specifiying them when starting your client:
+```
+geth --bootnodes "enode://<node ID>@<IP address>:<port>"
+```
+
+## Run a bootnode {#run-a-bootnode}
+
+Bootnodes are full nodes that are not behind a NAT. Every full node can act as a bootnode as long as it is publicly available.
```
Comment (jmcook1186) [inline src/content/bootnodes/index.md:22]:
```suggestion
Bootnodes are full nodes that are not behind a NAT ([Network Address Translation](https://www.geeksforgeeks.org/network-address-translation-nat/)). Every full node can act as a bootnode as long as it is publicly available.
```

**Comment by quickchase** [comment]
Nice. Questions remaining in my head:

Is specifying a bootnode different than specifying a static peer? Is there a difference in the communication or behavior there? And does any client support simply being a bootnode (not a full node, i.e. It's only job is to bootstrap peering)?

Maybe not the place to discuss but also, who runs the current Bootnodes? Is there a governance process to include Bootnodes?

Cheers

**Comment by albertg361** [comment]
Keep up the good work team

File: src/content/bootnodes/index.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+When you start up a node it should log your enode, which is a public identifier that others can use to connect to your node.
+The enode is usually regenerated on every restart, so make sure to look at your clients documentation on how to generate a persistent enode for your bootnode.
```
Comment (pauljickling) [inline src/content/bootnodes/index.md:24]: nit: add

Comment (jmcook1186) [reply src/content/bootnodes/index.md:24]:
```suggestion
The enode is usually regenerated on every restart, so make sure to look at your client's documentation on how to generate a persistent enode for your bootnode. 
```

**Comment by MariusVanDerWijden** [comment]
> Is specifying a bootnode different than specifying a static peer? Is there a difference in the communication or behavior there? 

Yes there's a difference, static nodes will be called over and over again while bootnodes will only be called if there are not enough peers to connect to (at least in geth)

> And does any client support simply being a bootnode (not a full node, i.e. It's only job is to bootstrap peering)?

Yes we do have a bootnode command in geth, but it kinda makes no sense really, so it is being phased out imo. 

> Maybe not the place to discuss but also, who runs the current Bootnodes? Is there a governance process to include Bootnodes?

The bootnodes in geth are run by the EF devops team and the geth team, there is no governance process to include/exclude them. 

cc @quickchase

File: src/content/bootnodes/index.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+title: Bootnodes
+description: The basic information you need to understand bootnodes
+lang: en
+---
+
+# Introduction to Ethereum Bootnodes {#bootnodes}
+
+## What are bootnodes? {#what-are-bootnodes}
+
+When a new node joins the Ethereum network it needs to connect to nodes that are already on the network in order to discover new nodes. These entrypoints into the Ethereum network are called bootnodes. Clients usually have a list of bootnodes built in that are run by the Ethereum Foundation or the client teams themselves.
```
Comment (jmcook1186) [inline src/content/bootnodes/index.md:11]:
```suggestion
When a new node joins the Ethereum network it needs to connect to nodes that are already on the network in order to discover new nodes. These entrypoints into the Ethereum network are called bootnodes. Clients usually have a list of bootnodes hardcoded into them. These bootnodes are typically run by the Ethereum Foundation's devops team or client teams themselves.
```

File: src/content/bootnodes/index.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Connect to a bootnode {#connect-to-bootnode}
+
+Most clients have a list of bootnodes build in, but you might want to connect to other bootnodes to connect by specifiying them when starting your client:
```
Comment (jmcook1186) [inline src/content/bootnodes/index.md:15]:
```suggestion
Most clients have a list of bootnodes build in, but you might also want to run your own bootnode, or use one that is not part of the client's hardcoded list. In this case, you can specifiy them when starting your client, as follows (example is for Geth, please check your client's documentation):
```

**Comment by jmcook1186** [comment]
moved the file into `/develoeprs/docs/nodes-and-clients` and added it to sidebar nav.
cc @minimalsm

File: src/content/developers/docs/nodes-and-clients/bootnodes/index.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+title: Introduction to Ethereum Bootnodes {#bootnodes}
+description: The basic information you need to understand bootnodes
+lang: en
+---
+
+
+When a new node joins the Ethereum network it needs to connect to nodes that are already on the network in order to discover new nodes. These entrypoints into the Ethereum network are called bootnodes. Clients usually have a list of bootnodes hardcoded into them. These bootnodes are typically run by the Ethereum Foundation's devops team or client teams themselves.
```
Comment (jmcook1186) [inline src/content/developers/docs/nodes-and-clients/bootnodes/index.md:8]:
```suggestion
When a new node joins the Ethereum network it needs to connect to nodes that are already on the network in order to then discover new peers. These entrypoints into the Ethereum network are called bootnodes. Clients usually have a list of bootnodes hardcoded into them. These bootnodes are typically run by the Ethereum Foundation's devops team or client teams themselves. Note that bootnodes are not the same as static nodes. Static nodes are called over and over again, whereas bootnodes are only called upon if there are not enough peers to connect to and a node needs to bootstrap some new connections.
```

**Review by corwintines — APPROVED** [review]
Thanks, @MariusVanDerWijden :)

**Pull request merged**